contk/metric/metric.py

```diff
@@ -61,8 +61,8 @@ def forward(self, data):
 			raise ValueError("Batch num is not matched.")
 
 		# perform random check to assert the probability is valid
-		checkid = random.randint(0, len(resp_length)-1)
```

Comment:
Why this two lines are changed?

Reply:
The previous random check method is not consistent with the data shape of gen_prob when resp_length < 2.

Reply:
There must be ```resp_length >= 2```, because `<go>` and `<eos>` is included. 
I don't think  `gen_prob.shape[1]-1` is right, because `gen_prob` never includes `<go>`. (Refer to the discription of function.) When `length=2`, checkrow must be `0`, which indicates `<eos>`

Another reason I used `resp_length[checkid]` is that I don't want to perform check on invalid positions, which is after the `<eos>`. We don't use the value on invalid positions to calculate perplexity, so there is no necessary to perform checks.